viewing_party/main.py

```diff
@@ -0,0 +1,237 @@
+"""
+def get_unique_watched(user_data):
+    if not user_data["watched"]:
+        return []
+    
+    # get movies watched by user
+    # get movies watched by friends
+    # get unique movies the user has watched but friends have not watched.  
+    # compare movies watched by friends to amanda's watched
+    user_watched_movie_titles = user_data["watched"]
+    friends_watched_movie_titles = user_data["friends"]
+    user_unique_movies_titles = []
+    # for movie in amanda_watched_movie_titles:
+       
+    #FOR name_of_individual_thing_in_array IN array_you_want_to_loop_through:
+    for title in user_watched_movie_titles: 
+        count = 0 #unique movie is Zero
+        for friend in friends_watched_movie_titles: 
+            if count > 0:
+                break 
+            if title in friend["watched"]:
+                count += 1
+        if count == 0:    
+            user_unique_movies_titles.append(title)    
+     
+    return user_unique_movies_titles # list of dictionaries
+
+
+
+def get_watched_avg_rating(user_data):
+    #user_data = { 'watched': [{..},{..}] }
+    var_watched = user_data["watched"]
+    if user_data["watched"] == []:
+        return 0
+    #var_watched = [ {..} , {..} ]
+    rating_sum = 0.0
+    rating_avg = 0.0
+    
+    for element in var_watched: 
+        # element = {...} , {...} , {...} //Now you get each element in the list
+        #element = { 'genre': , 'rating':, 'title':}
+        # print("element:", element)
+        my_rating = element['rating'] # 4.8 or 2.0 or 3.9
+        
+        if my_rating == None:
+            my_rating = 0.0
+        # print("my_rating:",my_rating )
+        rating_sum += my_rating
+        # print("rating_sum:", rating_sum)
+    
+    rating_avg = rating_sum/len(var_watched)   
+    # print("rating_avg:", rating_avg) # 0
+    return rating_avg
+
+#does this parameter have to be the exact name
+def get_most_watched_genre(user_data):
+    # print(user_data)
+    if user_data["watched"] == []:
+        return None
+    movie_type = {}
+    for movie in user_data["watched"]:
+        genre = movie["genre"]
+        # print("genre:", genre)
+        if genre not in movie_type:
+            movie_type[genre] = 1
+        else:
+            movie_type[genre] +=1
+
+    print("$$$$$$$$$$$$$$:0", movie_type)
+     # [i]["genre"]
+    
+    # print(movie_type)
+    most_watched_genre_key = max(movie_type, key = movie_type.get)
+    print(most_watched_genre_key)
+    return most_watched_genre_key
+
+
+
+
+def get_friends_unique_watched(user_data):
+    movie_list_of_dict = []
+    user_movies = user_data["watched"]
+    friends = user_data["friends"]
+    for friend in friends:
+        movie_lst_friend_watched = friend["watched"]
+        for element in movie_lst_friend_watched:
+            # print()
+            if element not in user_movies and element not in movie_list_of_dict:
+                # {"title": "Title A"}
+                # print(f"{} is unique")
```

Comment:
comments like these are also good to remove before turning in the assignment in the future.